tango_simlib/tests/test_fandango_json_parser.py

```diff
@@ -0,0 +1,196 @@
+    'time': 1521724934.799581,
+    'unit': '',
+    'value': 'Device is OK',
+    'writable': 'READ'
+    }
+
+
+class GenericSetup(unittest.TestCase):
+    """A class providing the setUp method definition for the other test classes.
+    """
+    longMessage = True
+
+    def setUp(self):
+        super(GenericSetup, self).setUp()
+        self.fandango_json_file = pkg_resources.resource_filename(
+            'tango_simlib.tests.config_files', 'database2.json')
+        self.fandango_parser = fandango_json_parser.FandangoExportDeviceParser()
+        self.fandango_parser.parse(self.fandango_json_file)
+
+
+class TestFandangoJsonParser(GenericSetup):
+    """A test class that tests that the fandango json parser works correctly.
+    """
+    def test_preprocess_command_types(self):
+        """Testing that command property mapping was done well and command type
+        of mapped properties is of tango type.
+        """
+
+        # testing that the command properties have been renamed to
+        # match values in command signature
+        for cmd_name in self.fandango_parser.get_device_command_metadata():
+            self.assertEquals(
+                EXPECTED_CMD_PROPERTY_PARAMETERS.sort(),
+                self.fandango_parser.get_device_command_metadata()
+                [cmd_name].keys().sort(),
+                "The mapping of the command properties to the command"
+                "signature wasn't successful for '%s' command" % (cmd_name)
+                )
+            # testing that data type of cmd property values is a tango type and
+            # there is no occurence of 'Const' in the value of the cmd property
+            for cmd_prop_value in [
+                    self.fandango_parser.get_device_command_metadata()
+                    [cmd_name][INTERESTED_CMD_PROP[0]],
+                    self.fandango_parser.get_device_command_metadata()
+                    [cmd_name][INTERESTED_CMD_PROP[1]]]:
+                self.assertEquals(
+                    str(cmd_prop_value).find('Const'), -1,
+                    "The value (%s) of the cmd property, '%s', for the command, '%s',"
+                    "has 'Const'" % (cmd_prop_value, INTERESTED_CMD_PROP[0], cmd_name)
+                    )
+                self.assertEquals(
+                    EXPECTED_TANGO_TYPES[0], type(cmd_prop_value),
+                    "The data type (%s) of the cmd property, '%s', of the command, '%s',"
+                    "is not a tango type" % (cmd_prop_value, INTERESTED_CMD_PROP[0],
+                                             cmd_name)
+                    )
+
+    def test_preprocess_attribute_types(self):
+        """Testing values of attribute properties, 'data_type' and 'data_format',
+        of all attributes are changed to tango types.
+        """
+
+        # testing that data type of specified attr property values is a tango type
+        for attr_name in self.fandango_parser.get_device_attribute_metadata():
```

Comment:
Similar comments to the previous test apply here too.  We can probably do without `EXPECTED_TANGO_TYPES` and `INTERESTED_ATTR_PROP`.